src/LinkedGenome.py

```diff
@@ -0,0 +1,192 @@
+
+from __future__ import annotations
+from typing import (
+    Generic, TypeVar, Iterable,
+    Callable, Protocol
+)
+
+from collections import namedtuple
+
+from genome import Genome
+
+
+Feature = namedtuple("Feature", ["feature", "length"])
+
+T = TypeVar('T')
+
+class Link(Generic[T]):
+    """Doubly linked link."""
+
+    val: T
+    prev: Link[T]
+    next: Link[T]
+
+    def __init__(self, val: T, p: Link[T], n: Link[T]):
+        """Create a new link and link up prev and next."""
+        self.val = val
+        self.prev = p
+        self.next = n
+
+def insert_after(link: Link[T], val: T) -> None:
+    """Add a new link containing avl after link."""
+    new_link = Link(val, link, link.next)
+    new_link.prev.next = new_link
+    new_link.next.prev = new_link
+def insert_before(link: Link[T], val: T) -> None:
+    """Add a new link containing avl after link."""
+    new_link = Link(val, link, link.prev)
+    new_link.prev.next = new_link
+    new_link.next.prev = new_link
+
+class LinkedListGenome(Genome):
+    """
+    Representation of a genome.
+
+    Implements the Genome interface using linked lists.
+    """
+    head: Link[Feature]  # Dummy head link
+    empty_te, active_te, inactive_te = 0, 1, 2
+    active_identifier = {}
+    counter_te = 0
```

Comment:
You are declaring these variables global for the class. If you have two or more genomes (instances of the class) then they _share_ the variables. I will bet you good money that this is not what you want.

Reply:
I think I understand. Then, is it okay declaring the following variables as global  (mapping constant)?
``` python
empty_te, active_te, inactive_te = 0, 1, 2
```
And I guess the other variable should be `self.active_identifier` and `self.counter_te` respectively, isn't it?

Reply:
The constants should probably declared at the class level (I was a bit too quick to include that line), but the other two are specific to an instance, I would say.